tests/unit/test_crt.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# https://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import botocore.exceptions
+import pytest
+import s3transfer
+from botocore.compat import HAS_CRT
+from botocore.credentials import Credentials
+
+import boto3
+from boto3.s3.transfer import TransferConfig
+from tests import mock, requires_crt
+
+if HAS_CRT:
+    import awscrt.s3
+    from s3transfer.crt import BotocoreCRTCredentialsWrapper
+
+    import boto3.crt
+
+
+def create_test_client(service_name='s3', region_name="us-east-1"):
+    return boto3.client(
+        service_name,
+        region_name=region_name,
+        aws_access_key_id="access",
+        aws_secret_access_key="secret",
+        aws_session_token="token",
+    )
+
+
+USW2_S3_CLIENT = create_test_client(region_name="us-west-2")
+USE1_S3_CLIENT = create_test_client(region_name="us-east-1")
+
+
+class TestCRTTransferManager:
+    @requires_crt()
+    def test_create_crt_transfer_manager_with_lock_in_use(self):
+        with mock.patch('boto3.crt.acquire_crt_s3_process_lock') as lock:
+            lock.return_value = None
```

Comment:
Shouldn't we be patching the process lock for all of these test cases? It seems like there we could run into issues for some of these tests where a previous case acquired the lock and affect downstream tests.

Reply:
Added a fixture similar to the CLI one that will handle this for us. The only minor issue is the fixtures are evaluated at load time rather than test run time. That required adding a bailout with HAS_CRT to make it a no-op.